src/hugr/validate.rs

```diff
@@ -70,6 +70,48 @@ impl<'a> ValidationContext<'a> {
             self.validate_node(node)?;
         }
 
+        // Check that io nodes match their parents' signatures
+        self.validate_io_resources(self.hugr, self.hugr.root())?;
```

Comment:
This should rather go in `ops::validate::validate_io_nodes`.

We check the I/O signatures there, it _should_ be straightforward to also check the resources.